src/pages/guide/integration/data-fixtures-guide.md

```diff
@@ -0,0 +1,403 @@
+        'company'
+    )
+]
+```
+
+## Migration guide: Legacy to Modern
+
+If you have tests using legacy fixtures, you should migrate them to parametrized data fixtures with PHP Attributes. New legacy fixtures **cannot be created**.
+
+### Before (Legacy fixtures)
+
+```php
+/**
+ * @magentoDataFixture Magento/Catalog/_files/product_simple.php
+ */
+public function testProductPrice(): void
+{
+    $product = $this->productRepository->get('simple'); // Fixed SKU
+    $this->assertEquals(10, $product->getPrice()); // Fixed price
+}
+```
+
+### After (Parametrized data fixtures with PHP Attributes)
+
+```php
+#[
+    DataFixture(ProductFixture::class, ['price' => 10], 'product')
+]
+public function testProductPrice(): void
+{
+    $fixtures = DataFixtureStorageManager::getStorage();
+    $product = $fixtures->get('product');
+    $this->assertEquals(10, $product->getPrice());
+}
+```
+
+## Working with fixture data
+
+### Retrieving fixture data
+
+Always use `DataFixtureStorageManager` to retrieve fixture data created by fixtures:
+
+```php
+use Magento\TestFramework\Fixture\DataFixtureStorageManager;
+
+#[
+    DataFixture(ProductFixture::class, ['price' => 25.00], 'test_product')
+]
+public function testProduct(): void
+{
+    $product = DataFixtureStorageManager::getStorage()->get('test_product');
+    $this->assertEquals(25.00, $product->getPrice());
+}
+```
+
+## Complex example: Inventory with multiple sources
+
+Example showing fixture composition and references:
+
+```php
+use Magento\TestFramework\Fixture\DataFixtureStorageManager;
+
+#[
+    DataFixture(SourceFixture::class, ['source_code' => 'test_source'], 'source'),
+    DataFixture(StockFixture::class, as: 'stock'),
+    DataFixture(
+        StockSourceLinksFixture::class,
+        [['stock_id' => '$stock.stock_id$', 'source_code' => '$source.source_code$']]
+    ),
+    DataFixture(
+        StockSalesChannelsFixture::class,
+        ['stock_id' => '$stock.stock_id$', 'sales_channels' => ['base']]
+    ),
+    DataFixture(ProductFixture::class, as: 'p1'),
+    DataFixture(ProductFixture::class, as: 'p2'),
+    DataFixture(
+        SourceItemsFixture::class,
+        [
+            ['sku' => '$p1.sku$', 'source_code' => '$source.source_code$', 'quantity' => 10, 'status' => 1],
+            ['sku' => '$p2.sku$', 'source_code' => '$source.source_code$', 'quantity' => 20, 'status' => 1],
+        ]
+    ),
+    DataFixture(Indexer::class, as: 'indexer')
+]
+public function testInventoryConfiguration(): void
+{
+    $stock = DataFixtureStorageManager::getStorage()->get('stock');
+    $product1 = DataFixtureStorageManager::getStorage()->get('p1');
+    // Test inventory logic
+}
+```
+
+This example demonstrates:
+
+- Creating related fixtures (source, stock, products)
+- Linking fixtures together using `$alias.property$` references
+- Passing arrays of data with references
+- Building complex test scenarios with multiple dependencies
+
+## Best practices
+
```

Comment:
Adobe recommends that you follow these best practices when implementing fixtures.